hoot-core-test/src/test/cpp/hoot/core/TestUtils.cpp

```diff
@@ -266,4 +267,24 @@ void TestUtils::verifyStdMatchesOutputIgnoreDate(const QString stdFilePath,
   }
 }
 
+bool TestUtils::mkpath(const QString& path)
+{
+  //  Don't make it if it exists
+  if (QDir().exists(path))
+    return true;
+  //  Try to make the path 'retry' times waiting 'duration' microseconds in between tries
+  const int retry = 3;
+  //  100 msec should be enough to wait between tries
+  const unsigned int duration = 100000;
+  for (int i = 0; i < retry; i++)
+  {
+    if (QDir().mkpath(path))
+      return true;
+    usleep(duration);
+  }
+  //  Report failure
+  CPPUNIT_FAIL(QString("Couldn't create output directory: %1").arg(path).toStdString());
+  return false;
+}
+
```

Comment:
Function based on @mschicker's function `ImplicitTagRawRulesDeriverTest::makeOutDir` now in the `TestUtils` class to be used throughout the core tests.  Here I use `usleep` for millisecond granularity instead of second.  Repeat, say, 3 times if necessary.